packages/admin/cms-admin/src/contentScope/Provider.tsx

```diff
@@ -100,16 +109,15 @@ export function useContentScope<S extends ContentScopeInterface = ContentScopeIn
 
     const matchParamsString = JSON.stringify(match.params); // convert matchParams to string, like this we can memoize or callbacks more easily
     const scope = useMemo(() => parseScopeFromRouterMatchParams<S>(JSON.parse(matchParamsString)), [matchParamsString]);
-    const matchPath = match.path;
     const redirectPath = context.redirectPathAfterChange;
     const setScope = useCallback(
         (action: SetContentScopeAction) => {
             const newContentScope = action(scope);
             const pathAfterScopePath = redirectPath || "";
-            const url = generatePath(matchPath, formatScopeToRouterMatchParams(newContentScope));
+            const url = context.location.createUrl(newContentScope);
```

Comment:
I changed this because [path-to-regexp](https://www.npmjs.com/package/path-to-regexp) didn't work correctly with the paths array. Also, I seems more logical to use createUrl here as well (it has only been used for the URL for the default value before). I'm not sure if this could break something in projects. I tried it in the Demo and everything worked fine.